### PR TITLE
OLS-254: Refactor: store conversation history

### PR DESCRIPTION
**Pull request by tisnik**
## Description

[OLS-254](https://issues.redhat.com//browse/OLS-254): Refactor: store conversation history
+ new unit tests, as usual

## Type of change

- [x] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #[OLS-254](https://issues.redhat.com//browse/OLS-254)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.

### CODE REVIEW

File: tests/unit/app/endpoints/test_ols.py [hunk 1]
```diff
@@ -36,6 +36,50 @@ def test_retrieve_conversation_id_existing_id(load_config):
...
+    with pytest.raises(ValueError, match="Incorrect user ID"):
+        ols.store_conversation_history(user_id, conversation_id, llm_request, "")
+
+
+def test_store_conversation_history_improper_conversation_id(load_config):
+    """Test if basic input verification is done during history store operation."""
+    user_id = "1234"
+    conversation_id = "::::"
+    llm_request = LLMRequest(query="Tell me about Kubernetes")
+    with pytest.raises(ValueError, match="Incorrect conversation ID"):
```
Comment (bparees) [inline tests/unit/app/endpoints/test_ols.py:79]: "invalid", not "incorrect" (need to fix the error message)

File: tests/unit/app/endpoints/test_ols.py [hunk 2]
```diff
@@ -36,6 +36,50 @@ def test_retrieve_conversation_id_existing_id(load_config):
...
+        user_id, conversation_id, f"{query}\n\n{response}"
+    )
+
+
+def test_store_conversation_history_improper_user_id(load_config):
+    """Test if basic input verification is done during history store operation."""
+    user_id = "::::"
+    conversation_id = suid.get_suid()
+    llm_request = LLMRequest(query="Tell me about Kubernetes")
+    with pytest.raises(ValueError, match="Incorrect user ID"):
```
Comment (bparees) [inline tests/unit/app/endpoints/test_ols.py:70]: "invalid", not "incorrect" (need to fix the error message)

Comment (tisnik) [reply tests/unit/app/endpoints/test_ols.py:70]: yup, we'll first need to merge https://github.com/openshift/lightspeed-service/pull/407  then I'll update this one

File: tests/unit/app/endpoints/test_ols.py (hunk 1 above)
Comment (tisnik) [reply tests/unit/app/endpoints/test_ols.py:79]: yup, we'll first need to merge https://github.com/openshift/lightspeed-service/pull/407  then I'll update this one

Comment (tisnik) [reply tests/unit/app/endpoints/test_ols.py]: fixed @bparees

File: tests/unit/app/endpoints/test_ols.py (hunk 2 above)
Comment (tisnik) [reply tests/unit/app/endpoints/test_ols.py]: fixed @bparees

**Pull request merged**